src/fmu_interfaces/FMU.cpp

```diff
@@ -0,0 +1,221 @@
+/*****************************************************************
+ *  Copyright (c) Dassault Systemes. All rights reserved.        *
+ *  This file is part of FMIKit. See LICENSE.txt in the project  *
+ *  root for license information.                                *
+ *****************************************************************/
+
+#include <iostream>
+#include <stdlib.h>
+#include <stdio.h>
+#include <sstream> // stringstream
+#include <iomanip>
+#include <stdexcept> // for runtime_error
+
+#ifdef _WIN32
+#include <direct.h>
+#include "shlwapi.h"
+#pragma comment(lib, "shlwapi.lib")
+#else
+#include <stdarg.h>
+#include <dlfcn.h>
+#endif
+
+#include "FMU.h"
+
+using namespace std;
+using namespace fmikit;
+
+
+#define MAX_MESSAGE_SIZE 4096 // arbitrary
+#define INTERNET_MAX_URL_LENGTH 2083 // from wininet.h
+
+
+MessageLogger * FMU::m_messageLogger = nullptr;
+
+const char* FMU::platform() {
+#ifdef _WIN32
+
+#ifdef _WIN64
+	return "win64";
+#else
+	return "win32";
+#endif
```

Comment:
For 64 bit systems `_WIN32` and `_WIN64` are both defined?

Reply:
Oh, is this essentially just template code you took from elsewhere?

Reply:
yes